content/learn/book/the-game-loop/time-and-timers.md

```diff
@@ -0,0 +1,277 @@
+
+Note that Bevy's "fixed timesteps" are not the right mechanism to use for gameplay logic like "every 5 seconds update this building".
+Bevy only supports a single fixed timestep across your entire project, and its use is completely optional.
+Please see our [timers and cooldowns] section below for the tools available to model this type of behavior.
+
+[`Time<Fixed>::timestep`]: https://docs.rs/bevy/latest/bevy/prelude/struct.Time.html#method.timestep
+[`Main`]: https://docs.rs/bevy/latest/bevy/app/struct.Main.html
+[`FixedMain`]: https://docs.rs/bevy/latest/bevy/app/struct.FixedMain.html
+[`Real`]: https://docs.rs/bevy/latest/bevy/prelude/struct.Real.html
+[`Virtual`]: https://docs.rs/bevy/latest/bevy/prelude/struct.Virtual.html
+[`Fixed`]: https://docs.rs/bevy/latest/bevy/prelude/struct.Fixed.html
+[fix your timestep article]: https://gafferongames.com/post/fix_your_timestep/
+[`RunFixedMainLoop`]: https://dev-docs.bevy.org/bevy/app/struct.RunFixedMainLoop.html
+
+### Interpolation between ticks
+
+One key problem with using a fixed timestep is that your game logic (including physics!) will have an uneven
+number of updates between each frame.
+This will lead to visible jittering, lagginess and tiny speedups.
+
+To account for this, we need to distinguish between the logical and rendered position (and rotation, and sometimes scale) of
+game objects.
+This is a common problem and pattern for multiplayer games, but using a fixed timestep means that even single player
+games need to handle it.
+
+The core strategy is simple enough, but the devil is in the details:
+
+1. Create your own custom `GamePosition` type, holding the translation/rotation/scale information that your project needs.
+2. Modify and read this `GamePosition` type for all of your game logic and physics.
+3. Interpolate between the previous [`GlobalTransform`] and the `GamePosition`, based on the elapsed [`Fixed`] time.
+   1. This should occur after the final fixed update for the frame has run, but before rendering occurs.
+   2. You ultimately want to set the [`GlobalTransform`], but doing this correctly in the presence of hierarchies is hard.
+   3. As a result, modifying the local [`Transform`] and relying on transform propagation can be the least bad solution.
+
+Bevy does not currently offer any built-in functionality for this form of interpolation,
+but open source [ecosystem crates] are available to use and learn from.
+
+[ecosystem crates]: https://bevy.org/assets/
+[`GlobalTransform`]: https://docs.rs/bevy/latest/bevy/prelude/struct.GlobalTransform.html
+[`Transform`]: https://docs.rs/bevy/latest/bevy/prelude/struct.Transform.html
+
+## Timers and cooldowns
+
+Delta time is great for physics and animations,
+but what if we want to implement other time-driven gameplay logic?
+We might want to automatically generate cookie clicks every 5 seconds
+or add a 3 second cooldown for our fireball ability.
+
+The simplest of these tools is the [`Timer`].
```

Comment:
I've started diving into Bevy again recently and one thing that wasn't really clear to me when using `Timer`s is whether or not I am responsible for handling the case where a repeating timer elapses multiple times per tick, just how the fixed time system set might run multiple times to catch up. I'm not sure if it is appropriate to cover in this book section, but something to think about.

Reply:
That's a good note for the [`Timer`] docs.